### PR TITLE
fix: add support for microsoft onelake

### PR DESCRIPTION
**Pull request by vmuddassir-msft**

### CODE REVIEW

**Comment by vmuddassir-msft** [comment]
Made code change to test utils to handle abfss (abfss://@onelake.dfs.fabric.microsoft.com)
Reads/Writes need to use a fully qualified https or abfss (onelake.dfs.fabric.microsoft.com/ws-id/artifact-id

File: rust/src/storage/config.rs [hunk 1]
```diff
@@ -179,7 +181,7 @@ fn try_configure_azure(
     options: &StorageOptions,
 ) -> DeltaResult<Arc<DynObjectStore>> {
     let store = MicrosoftAzureBuilder::from_env()
-        .with_url(storage_url.as_ref())
+        .with_url(storage_url.as_ref())
```
Comment (watfordkcf) [inline rust/src/storage/config.rs:184]:
```suggestion
        .with_url(storage_url.as_ref())
```

File: rust/src/test_utils.rs [hunk 2]
```diff
@@ -33,8 +34,19 @@ impl IntegrationContext {
         // create a fresh bucket in every context. THis is done via CLI...
         let bucket = match integration {
             StorageIntegration::Local => tmp_dir.as_ref().to_str().unwrap().to_owned(),
-            _ => format!("test-delta-table-{}", Utc::now().timestamp()),
+            StorageIntegration::Onelake => {
+                let account_name = env::var("AZURE_STORAGE_ACCOUNT_NAME").unwrap_or(String::from("onelake"));
+                let container_name = env::var("AZURE_STORAGE_CONTAINER_NAME").unwrap_or(String::from("delta-rs"));
+                format!("{0}.dfs.fabric.microsoft.com/{1}", account_name,container_name)
```
Comment (watfordkcf) [inline rust/src/test_utils.rs:40]:
```suggestion
                format!("{0}.dfs.fabric.microsoft.com/{1}", account_name, container_name)
```

File: rust/src/test_utils.rs [hunk 3]
```diff
@@ -33,8 +34,19 @@ impl IntegrationContext {
...
+            },
+            StorageIntegration::OnelakeAbfs =>{
+                let account_name = env::var("AZURE_STORAGE_ACCOUNT_NAME").unwrap_or(String::from("onelake"));
+                let container_name = env::var("AZURE_STORAGE_CONTAINER_NAME").unwrap_or(String::from("delta-rs"));
+                format!("{0}@{1}.dfs.fabric.microsoft.com",container_name,account_name)
```
Comment (watfordkcf) [inline rust/src/test_utils.rs:45]:
```suggestion
                format!("{0}@{1}.dfs.fabric.microsoft.com", container_name, account_name)
```

**Comment by rtyler** [comment]
@vmuddassir-msft are you getting to a point to where you're ready to put this into Review?

**Comment by vmuddassir-msft** [comment]
> @vmuddassir-msft are you getting to a point to where you're ready to put this into Review?

Delta-rs needs to be upgraded to the latest version of [object_store v0.7.0](https://crates.io/crates/object_store/0.7.0) , in order to add Support for Microsoft Fabric / OneLake . I have upgraded the versions for object_store with other dependencies which reference object_store v0.6.1 

I have facing an issue after the upgrade :

image

This is because datafusion seems to be internally using the dependency crate arrow-array v45 while the latest version is arrow-array = { version = "46.0.0" }

image

**Comment by wjones127** [comment]
> This is because datafusion seems to be internally using the dependency crate arrow-array v45 while the latest version is arrow-array = { version = "46.0.0" }

Yes, we need to wait for datafusion to upgrade versions of arrow before we can. Can you upgrade object store independent of arrow?

**Comment by vmuddassir-msft** [comment]
> > This is because datafusion seems to be internally using the dependency crate arrow-array v45 while the latest version is arrow-array = { version = "46.0.0" }
> 
> Yes, we need to wait for datafusion to upgrade versions of arrow before we can. Can you upgrade object store independent of arrow?
Unable to upgrade Object store  independently as this is also a dependency for parquet and the latest version of paraquet v46 has linked dependencies to the latest versions of all arrow related crates which includes arrow-array

image

**Comment by wjones127** [comment]
@vmuddassir-msft understood. The next version of datafusion is out for vote, so it should be released sometime next week. Once that happens we should be able to get this PR in a merge-able state. :)

File: rust/src/writer/record_batch.rs [hunk 4]
```diff
@@ -372,7 +372,8 @@ pub(crate) fn divide_by_partition_values(
         })
         .collect::<Result<Vec<_>, DeltaWriterError>>()?;
 
-    let partition_ranges = lexicographical_partition_ranges(sorted_partition_columns.as_slice())?;
```
Comment (vmuddassir-msft) [inline rust/src/writer/record_batch.rs:375]: #[allow(warnings)]  for now as [:lexicographical_partition_ranges ](https://arrow.apache.org/rust/arrow/compute/fn.lexicographical_partition_ranges.html) is Deprecated

Comment (vmuddassir-msft) [reply rust/src/writer/record_batch.rs:375]: @wjones127 / @rtyler  - Datafusion packages have been upgraded

**Comment by vmuddassir-msft** [comment]
> @vmuddassir-msft are you getting to a point to where you're ready to put this into Review?

@rtyler - PR is ready to be reviewed

**Review by rtyler — DISMISSED** [review]

**Comment by rtyler** [comment]
@vmuddassir-msft is your expectation that the test will never be run? (in which case we should just delete it), or do you want to configure a bearer token in GitHub secrets so that at least CI can run?

File: rust/src/writer/record_batch.rs (hunk 4 above)
Comment (wjones127) [reply rust/src/writer/record_batch.rs:375]: FYI another dedicated PR was merged to upgrade datafusion, so you'll need to address conflicts with that. Should be quick.

**Comment by rtyler** [comment]
Closing this in favor of the rebased #1642

**Pull request closed without merging**